quark/core/shurikenapkinfo.py

```diff
@@ -0,0 +1,612 @@
+        """
+        return {
+            method for method in self.all_methods if not method.cache.external
+        }
+
+    @property
+    def all_methods(self) -> Set[MethodObject]:
+        """
+        Inherited from baseapkinfo.py.
+        Return all methods including Android native APIs and custom methods
+        declared in the sample.
+
+        :return: a set of MethodObjects
+        """
+        methods = set()
+
+        match self.ret_type:
+            case "APK":
+                numOfMethod = (
+                    self.analysis.get_number_of_methodanalysis_objects()
+                )
+                for i in range(numOfMethod):
+                    methodAnalysis = self.analysis.get_analyzed_method_by_idx(
+                        i
+                    )
+                    methods.add(
+                        self.__convert_to_method_object(methodAnalysis)
+                    )
+
+            case "DEX":
+                # TODO - Wait for the upstream to support getting all methods.
+                for i in range(self.analysis.get_number_of_classes()):
+                    rawClass = self.analysis.get_class_by_id(i)
+                    className = rawClass.class_name.decode()
+                    classAnalysis = self.analysis.get_analyzed_class(className)
+                    for j in range(classAnalysis.n_of_methods):
+                        methodAnalysis = classAnalysis.methods[j].contents
+                        method = self.__convert_to_method_object(
+                            methodAnalysis
+                        )
+                        lowerMethodInfo = self.lowerfunc(method)
+                        lowerMethods = [info[0] for info in lowerMethodInfo]
+                        methods = methods.union(set(lowerMethods))
+                        methods.add(method)
+            case _:
+                raise ValueError("Unsupported File type.")
+
+        return methods
+
+    @functools.lru_cache()
+    def find_method(
+        self,
+        class_name: Optional[str] = None,
+        method_name: Optional[str] = None,
+        descriptor: Optional[str] = None,
+    ) -> List[MethodObject]:
+        """
+        Inherited from baseapkinfo.py.
+        Find a method with the given class name, method name, and descriptor.
+
+        :param class_name: the class name of the target method. Defaults to
+        None
+        :param method_name: the method name of the target method. Defaults to
+        None
+        :param descriptor: the descriptor of the target method. Defaults to
+        None
+        :return: a list of the target MethodObject
+        """
+        methods = self.all_methods
+        if class_name:
+            methods = (m for m in methods if class_name == m.class_name)
+
+        if method_name:
+            methods = (m for m in methods if method_name == m.name)
+
+        if descriptor:
+            methods = (m for m in methods if descriptor == m.descriptor)
```

Comment:
It might be better to make all if statements into one line:

```
methods = (m for m in methods if(class_name == m.class_name 
                                 and method_name == m.name 
                                 and descriptor == m.descriptor))
```